src/plugins/graph/components/legend/layer-legend.tsx

```diff
@@ -148,3 +147,49 @@ export const LayerLegend = observer(function LayerLegend(props: ILayerLegendProp
     </>
   );
 });
+
+export const LayerLegend = observer(function LayerLegend(props: ILegendPartProps) {
+  const graphModel = useGraphModelContext();
+  return (
+    <>
+      {
+        graphModel.layers.map((layer) => {
+          return (
+            <DataConfigurationContext.Provider key={layer.id} value={layer.config}>
+              <SingleLayerLegend {...props} />
+            </DataConfigurationContext.Provider>);
+          }
+        )
+      }
+    </>
+  );
+});
+
+function heightOfOneLayerLegend(layer: IGraphLayerModel) {
+  if (!layer.config.dataset) return 0;
+
+  const yAttrDescriptions = layer.config.yAttributeDescriptions.length;
+  // Only include the add series button if we have unused attributes
+  const attributeCount = layer.config.dataset?.attributes?.length ?? 0;
+  const addSeriesButton = (yAttrDescriptions + 1) < attributeCount ? 1 : 0;
+  const rows = Math.ceil((yAttrDescriptions + addSeriesButton) / 2);
+  return kLayerLegendHeaderHeight + kLayerLegendRowHeight * rows;
+}
+export function heightOfLayerLegend({ graphModel }: ILegendHeightFunctionProps) {
+  return graphModel.layers.reduce((prev, layer) => {
+    return prev + heightOfOneLayerLegend(layer);
+  }, 0);
+}
+
+export const colorIdsOfLayerLegend: ColorIdListFunction = function colorIdsOfLayerLegend(graphModel) {
+  let ids: string[] = [];
+  const sharedModelManager = getSharedModelManager(graphModel);
+  if (sharedModelManager?.isReady) {
+    const sharedDataSets =
+      sharedModelManager.getTileSharedModelsByType(graphModel, SharedDataSet) as SharedDataSetType[];
+    sharedDataSets.forEach(dataSet => {
+      ids = ids.concat(Array.from(dataSet.dataSet.attributes).map(attr => attr.id));
```

Comment:
It would be easier to iterate through `graphModel.layers` rather than going through sharedModelManager, I think